### PR TITLE
Fuzz tests for the RHost Pipes

### PR DESCRIPTION
**Pull request by karthiknadig**

### CODE REVIEW

**Comment by MikhailArkhipov** [comment]
👍

File: src/Host/Protocol/Test/RHostPipe/RHostPipeTest.cs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+            process.Start();
+            process.WaitForExit(250);
+            if (process.HasExited && process.ExitCode < 0) {
+                return;
+            }
+
+            CancellationTokenSource cts = new CancellationTokenSource(3000);
+            Task.Run(() => {
+                cts.Token.WaitHandle.WaitOne();
```
Comment (AlexanderSher) [inline src/Host/Protocol/Test/RHostPipe/RHostPipeTest.cs]: What is this code for? Killing process after 3 seconds?

File: src/Host/Protocol/Test/RHostPipe/RHostPipeTest.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            return "{[" + string.Join(",", parts) + "]}";
+        }
+
+        private byte[] GenerateBytes(int max = 1000000) {
+            byte[] rbuf = new byte[_rand.Next(0, max)];
+            _rand.NextBytes(rbuf);
+            return rbuf;
+        }
+
+        private ulong GenerateUInt64() {
```
Comment (AlexanderSher) [inline src/Host/Protocol/Test/RHostPipe/RHostPipeTest.cs]: These methods can be useful in other tests. You may want to add `Random` class to the `Microsoft.UnitTests.Core` assembly.

File: src/Host/Protocol/Test/RHostPipe/RHostPipeTest.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            
+            process.Start();
+            process.WaitForExit(250);
+
+            // Process should not exit before fully starting up
+            process.HasExited.Should().BeFalse();
+            Task killRhostTask = Task.CompletedTask;
+            Task readOutputTask = Task.CompletedTask;
+            try {
+                CancellationTokenSource cts = new CancellationTokenSource(3000);
```
Comment (AlexanderSher) [inline src/Host/Protocol/Test/RHostPipe/RHostPipeTest.cs]: `Task.Delay(3000).ContinueWith(t => TryKill(process));`

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**